### PR TITLE
solana: add min amount out to stage outbound

### PR DESCRIPTION
**Pull request by gator-boi**
Address AR finding 1.

### CODE REVIEW

File: solana/programs/swap-layer/src/processor/stage_outbound.rs
Comment (johnsaigle) [inline solana/programs/swap-layer/src/processor/stage_outbound.rs:76]: So`|| src_mint.key() == common::USDC_MINT` will short-circuit this check. Is this relying on other constraints where the src mint and dst mint can't both be equal to 0?

Comment (johnsaigle) [inline solana/programs/swap-layer/src/processor/stage_outbound.rs:76]: So|| src_mint.key() == common::USDC_MINT will short-circuit this check. Is this relying on other constraints where the src mint and dst mint can't both be equal to 0?

File: solana/programs/swap-layer/src/state/staged/outbound.rs
Comment (johnsaigle) [inline solana/programs/swap-layer/src/state/staged/outbound.rs:55]: I think this comment should either be deleted or expanded to explain what it does. Right now it's not adding anything

Comment (gator-boi) [reply solana/programs/swap-layer/src/state/staged/outbound.rs:55]: Fair enough, will remove.

File: solana/programs/swap-layer/src/processor/stage_outbound.rs
Comment (gator-boi) [reply solana/programs/swap-layer/src/processor/stage_outbound.rs:76]: We want to short circuit the check when the mint is USDC. The user doesn't need to specify a min_amount_out in that case, since there won't be a swap.

Comment (gator-boi) [reply solana/programs/swap-layer/src/processor/stage_outbound.rs:76]: This check guarantees that min_amount_out is some when the src_mint is not USDC. We don't really care if someone specifies a min_amount_out when transferring USDC, since that value will be ignored.

Comment (johnsaigle) [reply solana/programs/swap-layer/src/processor/stage_outbound.rs:76]: Hey sorry the original wording I had was very sloppy. Just edited it. Thanks for the clarification.

**Review by johnsaigle — APPROVED** [review]
This fixes the issue. 👍🏻 

The constraint leaves open the ability for a user to specify a `min_amount_out` even when this value will be ignored by the code. Enforcing a `None` value might be more correct and clear  - but also could cause a transaction to fail even when the system could complete it safely. 
For this reason I think this change can be merged as-is even if it might confuse someone down the line.

I'll defer to @mdulin2 as he's the one who identified the issue.

**Comment by mdulin2** [comment]
For safety, John is probably correct. But, in all practically, slippage means nothing with stable coins anyway. So, it's up to you all. 

The security issue is fixed though! That's the most important part.

**Review by a5-pickle — APPROVED** [review]

**Pull request merged**
